devito/types/dimension.py

```diff
@@ -848,13 +849,22 @@ def __init_finalize__(self, name, parent=None, factor=None, condition=None,
 
         super().__init_finalize__(name, parent)
 
-        self._factor = factor
+        # Always make the factor symbolic to allow overrides with different factor.
```

Comment:
We totally sure we don't want to leave this to the user? 

Probably yes for the principle of least surprise right? the underlying issue is when you pass an override with a TimeFunction with a different factor -- correct?

A symbolic factor will make the generated code a bit more verbose (thinking about when it's used in tandem with lazy streaming), but yeah, it sounds safer...

Reply:
I think it's safer to make it symbolic. It makes it a little it more verbose yes but at the price of avoiding bad surprises.